daml-foundations/daml-ghc/src/DA/Service/Daml/LanguageServer.hs

```diff
@@ -293,11 +293,11 @@ eventSlinger
 eventSlinger loggerH eventChan notifChan =
     forever $ do
         mbFatalErr <- atomically $ readTChan eventChan >>= \case
-            Compiler.EventFileDiagnostics (Compiler.Uri uri, diags) -> do
+            Compiler.EventFileDiagnostics (fp, diags) -> do
                 writeTChan notifChan
                     $ PublishDiagnostics
                     $ PublishDiagnosticsParams
-                    (Tagged uri)
+                    (Tagged $ Compiler.getUri $ Compiler.filePathToUri fp)
```

Comment:
Do we know that `fp` is always absolute? Why? If not, have you checked that the conversion logic on Windows doesn’t fall over?

Reply:
Yup, the only place this is called by is `updateFileDiagnostics`, my current PR is turning this FilePath into a Uri when you pass it in as it makes more sense

Reply:
I still don’t get it, why do we know that `updateFileDiagnostics` always uses an absolute path? Afaict, it just uses whatever is used as the key in Shake which is definitely not guaranteed to be absolute.

Reply:
Right you are, it should be `afp'` instead of `afp`

Reply:
That would only solve one direction, you would need to relativize when going the other direction. Otherwise you might end up with the same file represented twice by different keys which is going to be a mess. The reason why this works atm is probably because the IDE gets the initial filepath from VSCode which is absolute so we only ever use absolute filepaths.

This issue seems to have existed before, so I would just leave things for now (unless we see an actual issue) and rethink this carefully. It seems like we might want to move to URIs for keys in our Shake graph and then relativize at the edges where we care about it (at least when converting to DAML-LF but probably also in error messages printed to a terminal).

Reply:
https://github.com/digital-asset/daml/pull/901
This addresses one facet of the problem, I agree that shake keys should be Uris, in practice this gets a little hairy as some paths in the shake graphs really don't want to be made absolute